lib/matplotlib/artist.py

```diff
@@ -93,6 +100,46 @@ class Artist:
 
     zorder = 0
 
+    def __init_subclass__(cls):
+        # Inject custom set() methods into the subclass with signature and
+        # docstring based on the subclasses' properties.
+        # docstring based on the subclasses' properties.
+
+        if 'set' in cls.__dict__:
+            return  # don't overwrite set if the subclass defines set itself.
+
+        def set(self, **kwargs):
+            Artist.set(self, **kwargs)
+
+        cls.set = set
+        set.__doc__ = ("Set properties.\n\nSupported properties are:\n\n"
```

Comment:
Keep the "batch" language from the base-class?

Reply:
Changed to "Set multiple properties at once.". I have the feeling that "batch" is too much jargon.